pkgs/games/anki/default.nix

```
@@ -1,199 +1,372 @@
+    sha256 = "sha256-bAtmMGWi5ETIidFFnG3jzJg2mSBnH5ONO2/Lr9A3PpQ=";
+  };
+
+  # https://discourse.nixos.org/t/mkyarnpackage-lockfile-has-incorrect-entry/21586/3
+  anki-nodemodules = stdenv.mkDerivation {
+    pname = "anki-nodemodules";
+    inherit version src yarnOfflineCache;
+
+    nativeBuildInputs = [
+      fixup_yarn_lock
+      yarn
+      nodejs-slim
+    ];
+
+    configurePhase = ''
+      export HOME=$NIX_BUILD_TOP
+      yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+      fixup_yarn_lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+      yarn run postinstall --offline
+    '';
+
+    installPhase = ''
+      mv node_modules $out
+    '';
+  };
+
+  # Mask ninja's setup hooks, we just want the binary, not the implicit 'ninja install'
+  ninja-bin = buildEnv {
+    name = ninja.name;
+    paths = [ ninja ];
+    pathsToLink = [ "/bin" ];
+  };
+in
+python39.pkgs.buildPythonApplication {
+  inherit pname version src;
+
   outputs = [ "out" "doc" "man" ];
 
-  propagatedBuildInputs = [
-    pyqtwebengine
-    sqlalchemy
+  patches = [
+    ./gl-fixup.patch
+    ./no-update-check.patch
+  ];
+
+  buildInputs = [
+    anki-rsbridge
+    qt6.qtbase
+  ];
+  nativeBuildInputs = [
+    fakeCargo
+    fakeCorepack
+    fakeGit
+    fixup_yarn_lock
+    offlineYarn
+
+    ninja-bin
+    qt6.wrapQtAppsHook
+    rsync
+  ];
+  nativeCheckInputs = with python39.pkgs; [ pytest mock astroid  ];
+  propagatedBuildInputs = with python39.pkgs; [
+    # This rather long list came from running:
+    #    grep --no-filename -oE "^[^ =]*" python/{requirements.base.txt,requirements.bundle.txt,requirements.qt6_4.txt} | \
+    #      sort | uniq | grep -v "^#$"
+    # in their repo at the git tag for this version
+    # There's probably a more elegant way, but the above extracted all the
+    # names, without version numbers, of their python dependencies. The hope is
+    # that nixpkgs versions are "close enough"
+    # I then removed the ones the check phase failed on (pythonCatchConflictsPhase)
     beautifulsoup4
-    send2trash
-    pyaudio
-    requests
+    certifi
+    charset-normalizer
+    click
+    colorama
     decorator
-    markdown
+    distro
+    flask
+    flask-cors
+    idna
+    importlib-metadata
+    itsdangerous
+    jinja2
     jsonschema
-    setuptools
-  ]
-  ++ lib.optional plotsSupport matplotlib
-  ++ lib.optionals stdenv.isDarwin [ CoreAudio ]
-  ;
+    markdown
+    markupsafe
+    orjson
+    pep517
+    python39.pkgs.protobuf
+    pyparsing
+    pyqt6
+    pyqt6-sip
+    pyqt6-webengine
+    pyrsistent
+    pysocks
+    requests
+    send2trash
+    six
+    soupsieve
+    urllib3
+    waitress
+    werkzeug
+    zipp
+  ] ++ lib.optionals stdenv.isDarwin [ CoreAudio ];
 
-  nativeCheckInputs = [ pytest glibcLocales mock nose ];
+  # Activate optimizations
+  RELEASE = "1";
```

Environment variables are `toString`ed anyway. `"1"`, `1` and `true` have no difference.
```suggestion
  RELEASE = true;
```